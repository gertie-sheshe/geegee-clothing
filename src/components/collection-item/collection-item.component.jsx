import React from 'react';
import {connect} from 'react-redux';

import CustomButton from '../custom-button/custom-button.component';
import { addItemToCart } from '../../redux/cart/cart.actions.js';

import './collection-item.styles.scss';

const CollectionItem = ({item, addItem}) => {
    const { imageUrl, name, price } = item
    return (
    <div className='collection-item'>
        <div style={{
            backgroundImage: `url(${imageUrl})`
        }} className='image' />

        <div className='collection-footer'>
            <span className='name'>{name}</span>
            <span className='price'>{price}</span>
        </div>
        <CustomButton onClick={() => addItem(item)} inverted> ADD TO CART </CustomButton>
    </div>
)};

const mapDispatchToProps = dispatch => ({
    addItem: item => dispatch(addItemToCart(item))
});

export default connect(null, mapDispatchToProps)(CollectionItem);