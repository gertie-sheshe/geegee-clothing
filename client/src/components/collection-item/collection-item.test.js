import React from 'react';
import { shallow } from 'enzyme';

import { CollectionItem } from './collection-item.component';

describe('Collection Item Tests', () => {
  let wrapper;
  let mockAddItem;
  const imageUrl = 'www.testImage.com';
  const mockName = 'black hat';
  const mockPrice = 10;

  beforeEach(() => {
    mockAddItem = jest.fn();

    const mockProps = {
      item: {
        imageUrl: imageUrl,
        price: mockPrice,
        name: mockName,
      },
      addItem: mockAddItem,
    };

    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  it('Should render CollectionItem component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call addItem when AddButton is clicked', () => {
    wrapper.find('AddButton').simulate('click');
    expect(mockAddItem).toHaveBeenCalled();
  });

  it('Should render imageUrl as a prop on Background Image', () => {
    expect(wrapper.find('BackgroundImage').prop('imageUrl')).toBe(imageUrl);
  });

  it('Should render name prop in NameContainer', () => {
    expect(wrapper.find('NameContainer').text()).toBe(mockName);
  });

  it('Should render price prop in PriceContainer', () => {
    const price = parseInt(wrapper.find('PriceContainer').text());
    expect(price).toBe(mockPrice);
  });
});
