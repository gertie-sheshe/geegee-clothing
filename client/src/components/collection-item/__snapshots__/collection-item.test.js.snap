// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection Item Tests Should render CollectionItem component 1`] = `ShallowWrapper {}`;
